luigi/contrib/k8s_job.py

```diff
@@ -0,0 +1,193 @@
+
+
+"""
+Kubernetes Job wrapper for Luigi.
+
+From the Kubernetes website:
+
+    Kubernetes is an open-source system for automating deployment, scaling,
+    and management of containerized applications.
+
+For more information about Kubernetes Jobs: http://kubernetes.io/docs/user-guide/jobs/
+
+Requires:
+
+- pykube: ``pip install pykube``
+
+Written and maintained by Marco Capuccini (@mcapuccini).
+"""
+
+import luigi
+from luigi import configuration
+import logging
+import uuid
+import time
+
+try:
+    from pykube.config import KubeConfig
+    from pykube.http import HTTPClient
+    from pykube.objects import Job
+except ImportError:
+    logger.warning('pykube is not installed. KubernetesJobTask requires pykube.')
+
+class KubernetesJobTask(luigi.Task):
+
+    __POLL_TIME = 5 # see __track_job
+
+    def __init__(self, *args, **kwargs):
+        super(KubernetesJobTask, self).__init__(*args, **kwargs)
+        self.__logger = logging.getLogger('luigi-interface')
+        self.__logger.debug("Kubernetes auth method: " + self.auth_method)
+        if(self.auth_method == "kubeconfing"):
+            self.__kube_api = HTTPClient(KubeConfig.from_file(self.kubeconfig_path))
+        elif(self.auth_method == "ServiceAccount"):
+            self.__kube_api = HTTPClient(KubeConfig.from_service_account())
+        else:
+            raise ValueError("Illegal auth_method")
+        self.job_uuid = str(uuid.uuid4().hex)
+        self.uu_name = self.name + "-luigi-" + self.job_uuid
+        if ("restartPolicy" not in self.spec_schema):
+            self.spec_schema["restartPolicy"] = "Never"
+
+    @property
+    def auth_method(self):
+        """
+        This can be set to ``kubeconfing`` or ``ServiceAccount``.
+        It defaults to ``kubeconfing``.
+
+        For more details please referer to:
+
+        - kubeconfing: http://kubernetes.io/docs/user-guide/kubeconfig-file
+        - ServiceAccount: http://kubernetes.io/docs/user-guide/service-accounts
+        """
+        return configuration.get_config().get("k8s", "auth_method", "kubeconfing")
+
+    @property
+    def kubeconfig_path(self):
+        """
+        Path to kubeconfing file, for cluster authentication.
+        It defaults to "~/.kube/config", which is the default location
+        when using minikube (http://kubernetes.io/docs/getting-started-guides/minikube).
+        When auth_method is ``ServiceAccount`` this properity is ignored.
+
+        **WARNING**: For Python versions < 3.5 kubeconfing must point to a Kubernetes API
+        hostname, and NOT to an IP address.
+
+        For more details please referer to:
+        http://kubernetes.io/docs/user-guide/kubeconfig-file
+        """
+        return configuration.get_config().get("k8s", "kubeconfig_path", "~/.kube/config")
+
+    @property
+    def name(self):
+        """
+        A name for this job. This task will automatically append a UUID to the
+        name before to submit to Kubernetes.
+        """
+        pass
```

Comment:
I would probably also have this as a `raise NotImplementedError("subclass must define name")` or you can keep both this and the one below as `pass`.